### PR TITLE
Adding loader for Hainsworth dataset 

### PR DESCRIPTION
**Pull request by tanmayy24**
Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the size of the datasets (e.g. number files and duration, hours)
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of music included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable) 

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `mirdata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in mirdata, e.g. `mirdata/my_dataset.py`
- [x] Create tests for your loader in `tests/datasets/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/mirdata.rst` and `docs/source/table.rst`
- [x] Run `black`, `flake8` and `mypy` (see [Running your tests locally](https://mirdata.readthedocs.io/en/stable/source/contributing.html?highlight=contributing#running-your-tests-locally)).
- [x] Run `tests/test_full_dataset.py` on your dataset.
- [x] Check that codecov coverage does not decrease.

If your dataset **is not fully downloadable** there are two extra steps you should follow:
- [x] Contacting the mirdata organizers by opening an issue or PR so we can discuss how to proceed with the closed dataset.
- [x] Show that the version used to create the checksum is the "canonical" one, either by getting the version from the dataset creator, or by verifying equivalence with several other copies of the dataset.
- [x] Make sure someone has run `pytest -s tests/test_full_dataset.py --local --dataset my_dataset` once on your dataset locally and confirmed it passes.

#### Please-do-not-edit flag
To reduce friction, we will make commits on top of contributor's pull requests by default unless they use the `please-do-not-edit` flag. If you don't want this to happen don't forget to add the flag when you start your pull request.

### CODE REVIEW

File: mirdata/datasets/hainsworth.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+    Returns:
+        BeatData: beat annotations
+
+    """
+    beat_times = []
+    beat_positions = []
+
+    reader = csv.reader(fhandle, delimiter="\t")
+    for line in reader:
```
Comment (guillemcortes) [inline mirdata/datasets/hainsworth.py:178]: I guess that in Hainsworth all files are properly defined as tsv, not like in Ballroom (https://github.com/mir-dataset-loaders/mirdata/blob/master/mirdata/datasets/ballroom.py#L197)

**Review by guillemcortes — CHANGES_REQUESTED** [review]
Minor changes requested :)

File: mirdata/datasets/hainsworth.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+.. admonition:: Dataset Info
+    :class: dropdown
+
+    **Dataset Overview:**
+
+    The Hainsworth Dataset comprises 222 musical excerpts, each approximately 1 minute in length, categorized into six genres: rock/pop, dance, jazz, folk, classical, and choral. It was created by Stephen Hainsworth as part of his PhD thesis on automatic music transcription. The dataset offers annotations for beat and downbeat locations, which were generated in a two-stage process. Initially, initial taps were recorded, and then annotations were manually corrected using a custom interface in Matlab, guided by a time-frequency representation.
+
+    Of particular significance is the inclusion of approximately 20 choral examples, which posed a significant challenge for annotation due to their unique characteristics. This dataset gained recognition within the beat tracking community for its contribution to annotating and analyzing such challenging musical signals.
+
+    In 2014, Böck et al. [BockKW14b] conducted revisions on the beat and downbeat annotations to correct errors, leading to an enhancement in performance.
```
[the remark this answers is not in the document]
Comment (tanmayy24) [reply mirdata/datasets/hainsworth.py:12]: done

File: mirdata/datasets/hainsworth.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+    **Applications:**
+
+    The Hainsworth Dataset Loader is valuable for tasks related to beat tracking, rhythm analysis, and downbeat detection in various musical genres. Researchers and developers can utilize this dataset for algorithm development, testing, and evaluation. Additionally, it serves as a valuable resource for educational purposes, providing insights into the rhythmic structures of different musical genres.
+
+    **Acknowledgments and References:**
+
+    We would like to acknowledge Stephen Hainsworth for creating this dataset and his significant contribution to the field of automatic music transcription. Special thanks to Böck et al. [BockKW14b] for their efforts in improving the dataset annotations.
```
[the remark this answers is not in the document]
Comment (tanmayy24) [reply mirdata/datasets/hainsworth.py:20]: done

File: mirdata/datasets/hainsworth.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
+"""Hainsworth Dataset Loader
+
+.. admonition:: Dataset Info
+    :class: dropdown
+
+    **Dataset Overview:**
+
+    The Hainsworth Dataset comprises 222 musical excerpts, each approximately 1 minute in length, categorized into six genres: rock/pop, dance, jazz, folk, classical, and choral. It was created by Stephen Hainsworth as part of his PhD thesis on automatic music transcription. The dataset offers annotations for beat and downbeat locations, which were generated in a two-stage process. Initially, initial taps were recorded, and then annotations were manually corrected using a custom interface in Matlab, guided by a time-frequency representation.
```
[the remark this answers is not in the document]
Comment (tanmayy24) [reply mirdata/datasets/hainsworth.py:8]: done

File: mirdata/datasets/hainsworth.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+}
+
+REMOTES = None
+
+LICENSE_INFO = (
+    "Creative Commons Attribution Non Commercial Share Alike 4.0 International."
+)
+
+DOWNLOAD_INFO = """
+    Unfortunately most of the Hainsworth dataset is not available for download.
```
[the remark this answers is not in the document]
Comment (tanmayy24) [reply mirdata/datasets/hainsworth.py:66]: done

File: mirdata/datasets/indexes/hainsworth_full_index_1.0.json
[the remark this answers is not in the document]
Comment (tanmayy24) [reply mirdata/datasets/indexes/hainsworth_full_index_1.0.json:1]: I still have light indexes left and will do it when i cover those up

File: mirdata/datasets/hainsworth.py [hunk 6]
```diff
@@ -0,0 +1,223 @@
+"""Hainsworth Dataset Loader
```
[the remark this answers is not in the document]
Comment (tanmayy24) [reply mirdata/datasets/hainsworth.py:1]: i didnt find any link for the dataset, the one i found doesnt open

File: mirdata/datasets/hainsworth.py (hunk 1 above)
Comment (tanmayy24) [reply mirdata/datasets/hainsworth.py:180]: Yes!

**Review by guillemcortes — APPROVED** [review]
all good, ready to merge :)

**Pull request merged**